Firestore/Swift/Source/Property Wrapper/FIRRequest.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  FIRRequest
+//
+//  Created by Florian Schweizer on 12.07.21.
+//
+
+import FirebaseFirestore
+import SwiftUI
+
+public enum FIRPredicate {
+    case isEqualTo(_ lhs: String, _ rhs: Any)
+
+    case isIn(_ lhs: String, rhs: [Any])
+    case isNotIn(_ lhs: String, _ rhs: [Any])
+
+    case arrayContains(_ lhs: String, _ rhs: Any)
+    case arrayContainsAny(_ lhs: String, _ rhs: [Any])
+
+    case isLessThan(_ lhs: String, _ rhs: Any)
+    case isGreaterThan(_ lhs: String, _ rhs: Any)
+
+    case isLessThanOrEqualTo(_ lhs: String, _ rhs: Any)
+    case isGreaterThanOrEqualTo(_ lhs: String, _ rhs: Any)
+}
+
+@available(iOS 13.0, *)
+public class FirebaseStore<T: Decodable>: ObservableObject {
+    @Published var items: [T] = []
+    
+    private let store = Firestore.firestore()
+    
+    init(_ collection: String, _ predicates: [FIRPredicate]) {
+        load(from: collection, withPredicates: predicates)
+    }
+    
+    private func load(from collection: String, withPredicates predicates: [FIRPredicate]) {
+        let reference = store.collection(collection)
```

Comment:
Note that a CollectionReference is a subclass of Query. You should be able to simplify your code if you treat this as a Query.

Reply:
Thanks for the hint, this really does simplify it a lot!